src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,122 @@
+
+    private DataModel model;
+    private UserSimilarity similarity;
+    private UserNeighborhood neighborhood;
+    private UserBasedRecommender recommender;
+    private List<RecommendedItem> recommendations;
+
+// se usa biMap porque tiene el metodo inverse()
+// que cambia llaves por valores. Util a la hora de buscar
+// uno de los elementos por su valor
+
+    private BiMap<String, String> users;
+    private BiMap<String, String> products;
+    private int totalReviews = 0;
+
+    public MovieRecommender(String pathToData) {
+
+        try {
+
+            InputStream fileStream = new FileInputStream(pathToData);
+            InputStream gzipStream = new GZIPInputStream(fileStream);
+            Reader decoder = new InputStreamReader(gzipStream);
+            BufferedReader br = new BufferedReader(decoder);
+            BufferedWriter bw = new BufferedWriter(new FileWriter("cleanedFile.csv"));
+
+            String ln;
+
+            users = HashBiMap.create();
+            products = HashBiMap.create();
+
+            String userID = "";
+            String productID = "";
+
+//          File crusher, procesa el texto a un csv con solo la informacion que queremos
+            System.out.println("Procesando el archivo...");
+            while ((ln = br.readLine()) != null) {
+
+                if (ln.contains("product/productId: ")) {
+                    productID = ln.split("product/productId: ")[1];
+                    if (!products.containsKey(productID)) {
+                        products.put(productID, products.size() + 1 + "");
+                    }
+                } else if (ln.contains("review/userId: ")) {
+                    userID = ln.split("review/userId: ")[1];
+
+                    if (!users.containsKey(userID)) {
+                        users.put(userID, users.size() + 1 + "");
+                    }
+                } else if (ln.contains("review/score: ")) {
+                    String score = ln.split("review/score: ")[1];
+                    bw.write(users.get(userID) + "," + products.get(productID) + "," + score + "\n");
+                    totalReviews++;
+                }
+            }
+            br.close();
+            bw.close();
+
+            model = new FileDataModel(new File("cleanedFile.csv"));
+            similarity = new PearsonCorrelationSimilarity(model);
+            neighborhood = new ThresholdUserNeighborhood(0.1, similarity, model);
+            recommender = new GenericUserBasedRecommender(model, neighborhood, similarity);
+
+        } catch(IOException e) {
+            System.err.println(e.getMessage());
+        } catch(TasteException e){
+            System.err.println(e.getMessage());
+        }
+
+    }
+
+    public List<String> getRecommendationsForUser(String userID) {
+
+        try {
+
+//            No se por que me pidio usar lonog, creo que por usar biMap...
+            recommendations = recommender.recommend(Long.parseLong(users.get(userID)), 3);
```

Comment:
I would recommend to research it and put your results :) You can check on the recommender API documentation